test/libsolidity/semanticTests/viaYul/memory_struct_allow.sol

```diff
@@ -0,0 +1,21 @@
+contract C {
+    struct S {
+        uint256 a;
+        uint256 b;
+    }
+
+    function f() public pure returns (uint256 a, uint256 b){
+        assembly {
+            // Make free memory dirty to check that the struct allocation cleans it up again.
+            let freeMem := mload(0x40)
+            mstore(freeMem, 42)
+            mstore(add(freeMem, 32), 42)
+        }
+        S memory s;
+        return (s.a, s.b);
```

Comment:
Note to self: Since struct is uninitialized `s.a` and `s.b` default to `0`. Actually here we are testing that they are not `42` (previous value at locations where `s.a` and `s.b` reside)!

Reply:
Exactly :-).